1/backend/src/resolvers.js

```diff
@@ -0,0 +1,86 @@
+const jwt = require('jsonwebtoken');
+const fs = require('fs');
+
+data = require('./data');
+
+let SECRET_KEY = fs.readFileSync('./src/key/secret.key', 'utf8');
+
+const resolvers = {
+    Query: {
+        todos: () => JSON.parse(JSON.stringify(data.todos))
+    },
+    Mutation: {
+        addTodo: (parent, args, context) => {
+            if(args.message != "" && args.message != null && context.token) {
+                let newTodo = {
+                    id: findNextId(),
+                    message: args.message,
+                    completed: false,
+                };
+                data.todos.push(newTodo);
+                return newTodo;
+            }
+            return;
+        },
+        deleteTodo: (parent, args, context) => {
+            if (args.id != null && context.token){
+                let index = data.todos.findIndex(td => td.id === parseInt(args.id));
+                if (index > 0){
+                    data.todos.splice( index, 1)
+                    return true
+                }
+                return false
+            }
```

Comment:
What do you return in the `else` branch?

I would always prefer guard clauses over nested if-clauses:
```js
const { id } = args
const { token } = context
if(!(id && token)) return null // or throw new Error('telling error message here')
// continue here - no unnecessary and hard to read if clauses
```